### PR TITLE
Implement signup form

### PR DESCRIPTION
**Pull request by danhemerlein**
This implements signup form functionality.  It adds `react-mail-chimp-subscribe` as a dependency.  

While we were able to get the submit button text to change to an error or success message, we had trouble getting it to revert back to saying just 'submit' after some amount of time.  This is due to the statues returned by `MailChimpSubscribe`

Screenshots of the present workaround error message: 

**Error**
image

### CODE REVIEW

File: src/components/SignUpForm/index.js [hunk 1]
```diff
@@ -4,47 +4,83 @@ import "what-input";
...
-  };
-
-  handleSubmit = event => {
-    event.preventDefault();
+    this.setState({ emailAddress: event.target.value });
   };
 
   render() {
+    const url = ENV.MAILCHIMP_URL;
+    const body = document.querySelector('body');
```
Comment (sepowitz) [inline src/components/SignUpForm/index.js]: I get why you did it this way, but the react way to do it is to use the ```onKeyUp``` prop, much like you did with with the onClick.

File: src/components/SignUpForm/index.js [hunk 2]
```diff
@@ -4,47 +4,83 @@ import "what-input";
...
-    this.setState({ value: event.target.value });
-  };
-
-  handleSubmit = event => {
-    event.preventDefault();
+    this.setState({ emailAddress: event.target.value });
   };
 
   render() {
+    const url = ENV.MAILCHIMP_URL;
```
Comment (sepowitz) [inline src/components/SignUpForm/index.js]: No need to store this in a const, as you're only using it once. You can just pass ```ENV.MAILCHIMP_URL``` directly into the url prop, or destructure it at the top of the file (after the import) like so: ```const { MAILCHIMP_URL } = ENV;```

**Comment by hhff** [comment]
So when I submit the form with the enter key, I get a slew of console logs:
image

It appears to me that the key press on Enter is firing the subscribe button dozens of times.

Whats the reason we didn't just keep this wrapped in a form element? Form Elements handle "enter as submit" natively without javascript

File: src/components/SignUpForm/index.js (hunk 1 above)
Comment (danhemerlein) [reply src/components/SignUpForm/index.js]: I fixed this by using the `form` element with an `onSubmit` and changing the `input` to a `button` with a `type="submit"` seemed like the most accessible solution, is that alright?

**Comment by danhemerlein** [comment]
@hhff I pushed some changes adding a `form` element and related functionality before I saw this comment, apologies, please let me know how it looks!!

File: src/components/SignUpForm/index.js [hunk 3]
```diff
@@ -4,47 +4,75 @@ import "what-input";
...
+              </div>
+            </form>
+            {status === 'sending' ? (
+              this.setState({
+                message: 'Loading...'
+              })
+            ) : null}
+            {status === 'success' ? (
+              this.setState({
+                message: 'Thanks for subscribing'
```
Comment (hhff) [inline src/components/SignUpForm/index.js]: Can we change to `Thank you!`

File: src/components/SignUpForm/index.js [hunk 4]
```diff
@@ -4,47 +4,75 @@ import "what-input";
...
+              })
+            ) : null}
+            {status === 'error' ? (
+              this.setState({
+                message: `Oops! Please try again`
```
Comment (hhff) [inline src/components/SignUpForm/index.js]: Can we change to `Please try again.`

**Comment by hhff** [comment]
superceded

**Pull request closed without merging**